### ISSUE TITLE
Puppeteer exit suddenly in page.goto 

### ISSUE THREAD

**Issue by ali-rnik** [puppeteer/puppeteer#3022]
My program has (an) issue(s), it ends the script inside page.goto(), and wont reached screenshot().
Any idea?

```
const puppeteer = require('puppeteer');

var args = process.argv
var exepath = '/usr/local/bin/chrome'
var url = args[2];
var imgname = args[3] + '.png'
var prox = '--proxy-server=127.0.0.1:' + args[4]
var igcer = '--ignore-certificate-errors'

if(args[5] == 'noproxy')
  prox = '';

console.log(url)
console.log(prox)
console.log(imgname)

puppeteer.launch({executablePath: exepath, args: [prox, igcer]}).then(async browser => 
{
  const page = await browser.newPage();

  await page.setRequestInterception(true);
  page.on('request', interceptedRequest => 
  {
    console.log("Real Request: " + interceptedRequest.url().toString());

    if(interceptedRequest.url().startsWith('https'))
    {
      var newurl = new URL(interceptedRequest.url().toString() + "\n");
      newurl.protocol = 'http:';
      console.log(" What sent: " + newurl.toString() + "\n");

      interceptedRequest.continue({
        url: newurl.toString(),
        method: 'GET',
        headers: {'X-Is-Https': 'true', 'Accept': '*/*', 'Connection': 'close'}
      });
    }
    else if(interceptedRequest.url().startsWith('http'))
    {
      console.log(" What sent: " + interceptedRequest.url().toString() + "\n");
      interceptedRequest.continue({
        url: interceptedRequest.url(),
        method: 'GET',
        headers: {'X-Is-Https': 'false', 'Accept': '*/*', 'Connection': 'close'}
      });
    }
    else
      interceptedRequest.abort();

  });

  page.on('requestfailed', interceptedRequest => 
  {
    console.log(interceptedRequest.url() + ' ' + interceptedRequest.failure().errorText);
    interceptedRequest.continue();
  });

  await page.goto(url,{timeout: 0, waitUntil: 'networkidle0'});
  await page.screenshot({path: imgname, fullPage: true});
  console.log("Reached Screenshot");
  await browser.close();

});
```

**Comment by aslushnikov** [comment]
This is a question about your program that happen to use puppeteer rather than puppeteer itself. Please use StackOverflow instead.

**Comment by ali-rnik** [comment]
My problem solved after I upgarded puppeteer to 1.6.x from 1.1.0.
